### PR TITLE
select fields for count in the paginate method

### PR DESCRIPTION
**Pull request by dmouse**
Hi!

Currently in the `Model::paginate()` method you can configure the columns to you want to return in the result but those columns are not used for the count operation.

image

In terms of improve the performance in the `count` my idea is add a new argument in the `paginate` method to configure separately the columns for the count and you can use some indexed fields to improve the performance.

### CODE REVIEW

File: src/Illuminate/Database/Eloquent/Builder.php [hunk 1]
```diff
@@ -799,13 +799,13 @@ public function pluck($column, $key = null)
      *
      * @throws \InvalidArgumentException
      */
-    public function paginate($perPage = null, $columns = ['*'], $pageName = 'page', $page = null)
+    public function paginate($perPage = null, $columns = ['*'], $pageName = 'page', $page = null, $count = ['*'])
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Builder.php:802]: missing phpdoc

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Database/Eloquent/Builder.php (hunk 1 above)
Comment (dmouse) [reply src/Illuminate/Database/Eloquent/Builder.php:803]: updated

**Comment by taylorotwell** [comment]
This is apparently a performance based PR but no performance benchmarks were included.

**Comment by dmouse** [comment]
@taylorotwell what kind of benchmark test do I need to include?

For example in this case, a query including all fields takes `125ms`
image

But if I count using an indexed field takes `4.43ms`
image

**Pull request closed without merging**